### PR TITLE
Feature/non cash cont

### PR DESCRIPTION
**Pull request by Gubbsy**
## Proposed changes

- Add non-cash-contributions section with:
  - Question page
  - Add page
  - Edit page
  - Summary Page

### CODE REVIEW

File: app/models/progress_and_spend/progress_update_non_cash_contribution.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  validate do
+
+    validate_length(
+      :description,
+      50,
+      I18n.t(
+        "activerecord.errors.models.progress_update_" \
+          "volunteer.attributes.description.too_long",
+        word_count: 50)
```
Comment (ptrelease) [inline app/models/progress_and_spend/progress_update_non_cash_contribution.rb:27]: Wrong translation

File: app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+<fieldset class="govuk-fieldset">
+ 	<legend class="govuk-fieldset__legend govuk-fieldset__legend--l">
+
+		<span class="govuk-caption-xl govuk-!-padding-bottom-2">
+			<%= t('progress_and_spend.progress_update.non_cash_contribution.non_cash_contribution_add_edit.page_caption') %>
+		</span>
+
+		<h1 class="govuk-heading-l govuk-!-padding-bottom-2">
+      <%= t('progress_and_spend.progress_update.non_cash_contribution.non_cash_contribution_add_edit.page_heading') %>
```
Comment (ptrelease) [inline app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb:19]: Text on rendered page is wrong.

Comment (ptrelease) [reply app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb:19]: And the Description of secured bit, needs to be H3 so looks bold.

Comment (ptrelease) [reply app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb:19]: Value of secured non-cash contribution - should be in H3 too

Comment (ptrelease) [reply app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb:19]: Button text is wrong

File: app/views/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,154 @@
+<%=
```
Comment (ptrelease) [inline app/views/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/show.html.erb:1]: Change and delete buttons should be uppercase

Comment (ptrelease) [reply app/views/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/show.html.erb:1]: Change button adds another

File: config/locales/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/en.yml [hunk 4]
```diff
@@ -7,8 +7,8 @@ en-GB:
           page_heading_one_non_cash_contribution: "You have added 1 non cash contribution"
           page_heading: "You have added %{non_cash_contribution_num} non cash contributions"
           page_caption: "Tell us how the project is going"
-          change_button: "change"
-          remove_button: "remove"
+          change_button: "Change"
+          remove_button: "Remove"
           add_another_non_cash_contribution_heading: "Do you need to add another secured non-cash contribution?"
```
Comment (ptrelease) [inline config/locales/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/en.yml:12]: there is a generic change (and I think remove) in the en.yml

File: app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb [hunk 5]
```diff
@@ -67,13 +67,13 @@
...
 
-    <div class="govuk-form-group govuk-!-margin-top-4 <%= "govuk-form-group--error" if
+    <div class="govuk-form-group govuk-!-margin-top-4  govuk-!-padding-bottom-6 <%= "govuk-form-group--error" if
       non_cash_contribution.errors['value'].any? %>">
 
       <%=
         f.label :value,
         t('progress_and_spend.progress_update.non_cash_contribution.non_cash_contribution_add_edit.amount'),
-        class: "govuk-label"
+        class: "govuk-heading-m"
```
Comment (ptrelease) [inline app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb:76]: If it looks great, ideal.  I've been using a small heading (aka h3).  But go with what matches design best.

**Review by ptrelease — APPROVED** [review]

File: config/locales/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/en.yml (hunk 4 above)
Comment (Gubbsy) [reply config/locales/funding_application/progress_and_spend/progress_update/non_cash_contribution/non_cash_contribution_summary/en.yml:12]: Yeah, this needs to be changed for the other pages we implemented too and made generic across the board. I will make a note on the tracker card and come back when we do our tidy up

File: app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb (hunk 5 above)
Comment (Gubbsy) [reply app/views/partials/funding_application/progress_and_spend/_add_non_cash_contribution.html.erb]: For consistency, I shall use S for h3

**Pull request merged**